src/main/java/de/tum/cit/aet/artemis/lecture/web/LectureResource.java

```diff
@@ -120,22 +123,51 @@ public LectureResource(LectureRepository lectureRepository, LectureService lectu
-        if (lecture.getId() != null) {
+    public ResponseEntity<SimpleLectureDTO> createLecture(@RequestBody SimpleLectureDTO newLectureDto) throws URISyntaxException {
+        log.debug("REST request to save Lecture : {}", newLectureDto);
+        if (newLectureDto.id() != null) {
             throw new BadRequestAlertException("A new lecture cannot already have an ID", ENTITY_NAME, "idExists");
         }
-        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, lecture.getCourse(), null);
+        Course course = courseRepository.findByIdElseThrow(newLectureDto.course.id());
+        authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, course, null);
+
+        Lecture newLecture = new Lecture();
+        updateLectureAttributesFromDTO(newLecture, newLectureDto);
+        newLecture.setCourse(course);
 
-        Lecture savedLecture = lectureRepository.save(lecture);
-        channelService.createLectureChannel(savedLecture, Optional.ofNullable(lecture.getChannelName()));
-        return ResponseEntity.created(new URI("/api/lecture/lectures/" + savedLecture.getId())).body(savedLecture);
+        Lecture savedLecture = lectureRepository.save(newLecture);
+        String channelName = channelService.createLectureChannel(savedLecture, Optional.ofNullable(newLectureDto.channelName()));
+        SimpleLectureDTO savedLectureDTO = SimpleLectureDTO.from(savedLecture, course, channelName);
+        return ResponseEntity.created(new URI("/api/lecture/lectures/" + savedLecture.getId())).body(savedLectureDTO);
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    public record SimpleLectureDTO(Long id, String title, String description, ZonedDateTime startDate, ZonedDateTime endDate,
+            @JsonProperty("isTutorialLecture") boolean isTutorialLecture, String channelName, CourseDTO course) implements LectureDTO {
+
+        public static SimpleLectureDTO from(Lecture lecture, String channelName) {
+            return from(lecture, lecture.getCourse(), channelName);
+        }
+
+        public static SimpleLectureDTO from(Lecture lecture, @NonNull Course course, @Nullable String channelName) {
+            return new SimpleLectureDTO(lecture.getId(), lecture.getTitle(), lecture.getDescription(), lecture.getStartDate(), lecture.getEndDate(), lecture.isTutorialLecture(),
+                    channelName, CourseDTO.from(course));
+        }
+
+        public record CourseDTO(Long id, String title, String studentGroupName, String teachingAssistantGroupName, String editorGroupName, String instructorGroupName) {
+
+            public static CourseDTO from(Course course) {
+                return new CourseDTO(course.getId(), course.getTitle(), course.getStudentGroupName(), course.getTeachingAssistantGroupName(), course.getEditorGroupName(),
+                        course.getInstructorGroupName());
+            }
+        }
     }
```

Comment:
Can we extract the DTO to the `/dto` folder to keep the resource short and clean?

Reply:
Also the DTO does not use any validation annotations which we usually do, the default is nullable

We might want to add @NotNull for `ìd` and probably @NotNull @Valid for `CourseDTO`?